### ISSUE TITLE
SSH: Create actual `Match` blocks

### ISSUE THREAD

**Issue by jficz** [nix-community/home-manager#2769]
### Description

Currently the option `programs.ssh.matchBlocks` only allows creating `Host` blocks but not `Match` blocks.

This is slightly confusing but the main issue is that there is no way (that I could find) to create `Match` blocks
except creating them manually.

`Match` is basically a superset of `Host` as it allows more options to be set besides the host name pattern (including
custom shell commands for instance). There is no difference in the content of the respective blocks so the implementing
logic is already there.

It would be great to be able to create `Match` blocks as well as `Host` blocks.

I've created a dumb prototype which tries to reuse the `Host` declaration - currently it's very simple ~and untested~ _and works well for the author_ : https://github.com/jficz/home-manager/tree/add-match-to-ssh

Any comments, tips or feedback welcome. No need to be PC, I can handle anything ;)

**Comment by cptMikky** [comment]
I'm confused: when I add my github branch as `home-manager` channel instead of upstream, I get this error:
```
error: The option `programs.ssh.matchBlocks.myserver.data.match' is used but not defined.
```

But when I add the local git repo to `NIX_PATH=(...):home-manager=/path/to/my/fork/home-manager` directly, it works.

**Comment by teto** [comment]
is the PR anywhere public ? could be interesting in case someone else want to help you

**Comment by sumnerevans** [comment]
@teto I think https://github.com/nix-community/home-manager/pull/2992 is what's being referred to

**Comment by jficz** [comment]
This can be closed when #2992 is merged.
